common/optimiser/OptimiserVariable.java

```diff
@@ -26,8 +26,7 @@
 
 public abstract class OptimiserVariable<T> {
 
-    protected T initial;
-    protected T solution;
+    protected T value;
```

Comment:
There's really no reason to keep the initial value separate from the solution value. We either want to keep the variable's value, or override it explicitly.